### ISSUE TITLE
rustc 1.80 fails building --locked from crates.io

### ISSUE THREAD

**Issue by jayvdb** [01mf02/jaq#198]
```
> rustc --version
rustc 1.80.0 (051478957 2024-07-21)
> cargo install --locked jaq
    Updating crates.io index
  Downloaded jaq v1.5.1
  Downloaded 1 crate (23.3 KB) in 1.26s
  Installing jaq v1.5.1
    Updating crates.io index
    Updating crates.io index
  Downloaded jaq-std v1.5.1
  Downloaded jaq-core v1.5.0
  Downloaded jaq-interpret v1.5.0
  Downloaded 3 crates (82.1 KB) in 1.16s
   Compiling version_check v0.9.4
   Compiling proc-macro2 v1.0.69
   Compiling unicode-ident v1.0.10
   Compiling quote v1.0.29
   Compiling libc v0.2.147
   Compiling serde v1.0.190
   Compiling cfg-if v1.0.0
   Compiling once_cell v1.18.0
   Compiling zerocopy v0.7.32
   Compiling allocator-api2 v0.2.16
   Compiling ahash v0.8.7
   Compiling proc-macro-error-attr v1.0.4
   Compiling syn v1.0.109
   Compiling proc-macro-error v1.0.4
   Compiling hashbrown v0.14.2
   Compiling itoa v1.0.9
   Compiling cc v1.0.79
   Compiling getrandom v0.2.10
   Compiling memchr v2.6.4
   Compiling syn v2.0.38
   Compiling serde_json v1.0.108
   Compiling chumsky v0.9.3
   Compiling aho-corasick v1.1.2
   Compiling equivalent v1.0.0
   Compiling libm v0.2.8
   Compiling libmimalloc-sys v0.1.35
   Compiling ryu v1.0.14
   Compiling regex-syntax v0.7.5
   Compiling indexmap v2.1.0
   Compiling heck v0.4.1
   Compiling log v0.4.20
   Compiling serde_derive v1.0.190
   Compiling os_str_bytes v6.4.1
   Compiling regex-automata v0.3.7
   Compiling hifijson v0.2.0
   Compiling dyn-clone v1.0.11
   Compiling time-core v0.1.0
   Compiling time v0.3.20
   Compiling clap_lex v0.3.2
   Compiling clap_derive v4.0.21
error[E0282]: type annotations needed for `Box<_>`
  --> /home/jayvdb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.20/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `jaq v1.5.1`, intermediate artifacts can be found at `/tmp/cargo-installKu45Io`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

installing without `--locked` works.
Solution would be to do a release with a new lock file

**Comment by 01mf02** [comment]
This error came already up once (#184). It is corrected in #199.

**Comment by jayvdb** [comment]
This issue is about crates.io , and cant be fixed unless there is a new release.

**Comment by 01mf02** [comment]
The issue also occurs when only running `cargo check` on the cloned repository, so it is not specific to crates.io.
So yes, the issue is gone locally, but I'm aware that it persists when running `cargo install jaq --locked`.
I will make a new release soon.

**Comment by 01mf02** [comment]
I just released jaq 1.6.0, which resolves this problem. Thanks again for reporting this!
